server/service.go

```diff
@@ -737,26 +727,19 @@ func (m *BoostService) processDenebPayload(w http.ResponseWriter, req *http.Requ
 				}
 			}
 
-			// Lock before accessing the shared payload
-			mu.Lock()
-			defer mu.Unlock()
-
-			if requestCtx.Err() != nil { // request has been cancelled (or deadline exceeded)
-				return
-			}
-
-			// Received successful response. Now cancel other requests and return immediately
 			requestCtxCancel()
-			*result = *responsePayload
+			resultCh <- responsePayload
 			log.Info("received payload from relay")
```

Comment:
I think there could be value in adding back a cancellation check here. There's also a check at the beginning of the function, but this one (closer to the end) is most likely to catch it, right? This would help prevent multiple, relatively verbose "received payload from relay" messages.

Reply:
I added back a check that should get rid of multiple received payload from relay messages, but its kinda ugly. I kinda prefer the multiple log lines, ptal